### ISSUE TITLE
[tests][nice-to-have] Resource Helpers would be more useful as builders

### ISSUE THREAD

**Issue by smklein** [oxidecomputer/omicron#2114]
Context: https://github.com/oxidecomputer/omicron/pull/2113/files#r1063578360

The resource helpers might be more useful if they used a "builder pattern", to allow optionally changing features, such as the authentication information across a wide range of `NexusRequest` objects

**Comment by davepacheco** [comment]
Yeah, they're a little long in the tooth.  At this point, if we were going to do something different, should we be using a progenitor-generated client instead?

**Comment by david-crespo** [comment]
Yes!

**Comment by davepacheco** [comment]
Another thing I've wondered about is whether we'd be better off using something like [Hurl](https://hurl.dev/index.html).  I think quite a lot of our happy-path testing and basic error testing could be described in terms of just a bunch of requests and expected responses (and a lot of it could even be automatically generated from the OpenAPI spec).  That said, the mechanism we use today also verifies a bunch of other useful properties (like that all 400- and 500-level response codes have a message and error code, that everything has a request id, etc.)
